src/gordon_gcp/clients/http.py

```diff
@@ -142,42 +142,50 @@ def __init__(self, auth_client=None, session=None):
             method=method.upper(),
             url=url,
             kwargs=req_kwargs))
-        async with self._session.request(
-                method, url, **req_kwargs) as resp:
-            log_kw = {
-                'request_id': request_id,
-                'method': method.upper(),
-                'url': resp.url,
-                'status': resp.status,
-                'reason': resp.reason
-            }
-            logging.debug(_utils.RESP_LOG_FMT.format(**log_kw))
-
-            if resp.status in REFRESH_STATUS_CODES:
-                logging.warning(f'[{request_id}] HTTP Status Code {resp.status}'
-                                f' returned requesting {resp.url}: '
-                                f'{resp.reason}')
-                if token_refresh_attempts:
-                    logging.info(f'[{request_id}] Attempting request to '
-                                 f'{resp.url} again.')
-                    return await self.request(
-                        method, url,
-                        token_refresh_attempts=token_refresh_attempts,
-                        request_id=request_id,
-                        **req_kwargs)
-
-                logging.warning(f'[{request_id}] Max attempts refreshing auth '
-                                f'token exhausted while requesting {resp.url}')
-
-            # avoid leaky abstractions and wrap http errors with our own
-            try:
-                resp.raise_for_status()
-            except aiohttp.ClientResponseError as e:
-                msg = f'[{request_id}] Issue connecting to {resp.url}: {e}'
-                logging.error(msg, exc_info=e)
-                raise exceptions.GCPHTTPResponseError(msg, resp.status)
-
-            return await resp.text()
+        try:
+            resp = await self._session.request(method, url, **req_kwargs)
+        except Exception as e:
+            msg = f'[{request_id}] Request call failed {resp.url}: {e}'
+            logging.error(msg, exc_info=e)
+            resp.close()
```

Comment:
do we really need to attempt to close this given the request raised an exception?

Reply:
Well, it might remove the secondary exception in the logs.  We don't super much care I guess, just seemed like a thing to do.  Although I suppose it could set off its own crap.

Reply:
Haven’t looked too closely at this patch yet, but doesn’t this need to close the session (rather than the response) since we’re no longer using the context manager? I would think it’s also a coroutine that needs to be awaited.

Reply:
Haven’t looked too closely at this patch yet, but doesn’t this need to close the session (rather than the response) since we’re no longer using the context manager? I would think it’s also a coroutine that needs to be awaited.

Reply:
We were kinda unsure exactly what the deal is with all the release() and close() methods all over the place.  But we can also just drop this and let it error on process cleanup.

Reply:
I don't think we really do, but why not be thorough?  There's other errors log a few lines down about an unclosed session when the program is shutting down, so at the very least it will make the log slightly cleaner.

```
gordon-janitor (1) ERROR: Unclosed connector
connections: ['[(<aiohttp.client_proto.ResponseHandler object at 0x7fcac0144940>, 1264.993380062)]']
connector: <aiohttp.connector.TCPConnector object at 0x7fcac21a23c8>
...
gordon-janitor (1) ERROR: Unclosed client session
client_session: <aiohttp.client.ClientSession object at 0x7fcac2135cf8>
```